### ISSUE TITLE
[Bug] Qwen decode padding races on inactive rows during dynamic batches

### ISSUE THREAD

**Issue by Little-oil** [hw-native-sys/pypto-serving#201]
## Diagnosis

**pypto-serving / Qwen decode runner.** The runner treats the compiled decode as fixed-batch and fills inactive rows by replicating active row 0. The Qwen decode ABI is runtime-batch dynamic, and native paged attention appends K/V in parallel. Replicated rows therefore make several lanes write the same physical KV slot concurrently.

## Description

With the PyPTO paged-attention implementation from hw-native-sys/pypto-lib#1031, long-context generation can silently produce corrupt output under chunked-prefill scheduling:

- 16 concurrent identical prompts, each 3338 tokens
- 128 greedy output tokens per request
- max sequence length 4096, chunked prefill enabled, prefix cache disabled
- the scheduler interleaves prefill and decode, so decode runs with partial runtime batches before reaching B16

Before the fix, one run had 1/16 streams diverge and a repeat had 2/16 streams containing only token ID 0. The process still exited successfully.

The inactive-row padding in `Qwen314BModelRunner._prepare_decode_inputs` copies row 0's token, sequence length, slot mapping, and block-table pages into every inactive row. The dispatch then exposes all max-batch rows. Although those duplicated values were intended to make the writes idempotent, parallel PA lanes writing the same K/V address are a race.

Expected behavior: decode should expose only `actual_batch` rows to the runtime-dynamic kernel. Inactive host-buffer capacity must not participate in the dispatch.

## Reproduction

```bash
pypto-serving \
  --model /data/models/Qwen3-14B \
  --platform a2a3 \
  --max-model-len 4096 \
  --max-num-seqs 16 \
  --max-num-batched-tokens 4096 \
  --long-prefill-token-threshold 2048 \
  --no-enable-prefix-caching \
  --enable-chunked-prefill \
  --generate-config '{"max_new_tokens":128,"temperature":0,"top_p":1,"ignore_eos":true}' \
  --prompt "<3338-token prompt>"  # submit 16 identical prompts
```

## Why pypto-lib's 40-layer validation does not catch it

1. The faulty padding is in pypto-serving, outside the PA/kernel implementation.
2. The lib validation uses a fixed synthetic invocation and does not exercise serving's partial runtime batches.
3. Under `--decode-steps > 1`, the lib test re-uploads host K/V each step, so device K/V writes do not persist; it also skips the host argmax check. It is a growing-context timing sweep, not true autoregressive generation.
4. The standalone PA test compares with `rtol=5e-3, atol=2e-2`; serving persists real prefill KV over 40 layers and 128 autoregressive steps.

## Proposed fix and validation

- Keep max-batch-capacity storage, but pass zero-offset prefix views for `seq_lens`, `slot_mapping`, `block_table`, logits, token IDs, sampled IDs, and next-hidden state.
- Handle both host tensors and worker-resident `DeviceTensor` sampling buffers.
- Stop writing replicated metadata into inactive rows.

Post-fix device regression:

- 16/16 requests completed with identical 128-token streams
- no all-zero or invalid stream
- 2048 aggregate generated tokens in 106.96 s (19.1 tok/s end-to-end)
- Qwen unit tests: 31 passed

A top-2 diagnostic also showed the synthetic prompt has an intrinsically near-tied greedy decision at generated token 6: PyPTO PA margin 0.01567; CCE margin 0.00096, and CCE can take the same alternate natural-language branch. Exact token identity across requests is therefore not a suitable numerical-correctness oracle here; the all-zero streams are the corruption signature fixed by removing duplicate KV writes.

## Environment

| Component | Revision/version |
|---|---|
| pypto-serving | `9db7e20e9fe65c359c9b9a7bed4d35764b7bbc9b` |
| pypto-lib PR | hw-native-sys/pypto-lib#1031, head `a523fa0ca68289ae91cc42d700ef41a54a3f971a` |
| pypto | `d2cf77c0d804dc1ded0af0c085cd3c66b4e96034` |
| simpler/runtime | `1f27a157828d021638531fe33fe194fb2632f4ff` |
| pto-isa | `f51c92f610827daad0ddfb383072e03d514b4ae9` |
| ptoas | 0.57 |
| CANN | 9.0.0 |
| Host | Linux aarch64 |
| Device platform | a2a3 |
| Model | Qwen3-14B |
